Extensions/PrimitiveDrawing/Extension.cpp

```diff
@@ -632,5 +633,148 @@ void DeclarePrimitiveDrawingExtension(gd::PlatformExtension& extension) {
+      .UseStandardParameters("number")
+      .MarkAsAdvanced();
+
+  obj.AddExpressionAndConditionAndAction("number",
+                "ScaleY",
+                _("Scale on Y axis"),
+                _("the height's scale of an object"),
+                _("the height's scale"),
+                _("Size"),
+                "res/actions/scaleHeight24.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .UseStandardParameters("number")
+      .MarkAsAdvanced();
+
+  obj.AddAction("FlipX",
+                _("Flip the object horizontally"),
+                _("Flip the object horizontally"),
+                _("Flip horizontally _PARAM0_ : _PARAM1_"),
+                _("Effects"),
+                "res/actions/flipX24.png",
+                "res/actions/flipX.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .AddParameter("yesorno", _("Activate flipping"))
+      .MarkAsSimple();
+
+  obj.AddAction("FlipY",
+                _("Flip the object vertically"),
+                _("Flip the object vertically"),
+                _("Flip vertically _PARAM0_ : _PARAM1_"),
+                _("Effects"),
+                "res/actions/flipY24.png",
+                "res/actions/flipY.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .AddParameter("yesorno", _("Activate flipping"))
+      .MarkAsSimple();
+
+  obj.AddCondition("FlippedX",
+                   _("Horizontally flipped"),
+                   _("Check if the object is horizontally flipped"),
+                   _("_PARAM0_ is horizontally flipped"),
+                   _("Effects"),
+                   "res/actions/flipX24.png",
+                   "res/actions/flipX.png")
+      .AddParameter("object", _("Object"), "Drawer");
+
+  obj.AddCondition("FlippedY",
+                   _("Vertically flipped"),
+                   _("Check if the object is vertically flipped"),
+                   _("_PARAM0_ is vertically flipped"),
+                   _("Effects"),
+                   "res/actions/flipY24.png",
+                   "res/actions/flipY.png")
+      .AddParameter("object", _("Object"), "Drawer");
+
+  obj.AddAction("Width",
+                _("Width"),
+                _("Change the width of an object."),
+                _("the width"),
+                _("Size"),
+                "res/actions/scale24.png",
+                "res/actions/scale.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .UseStandardOperatorParameters("number")
+      .MarkAsAdvanced();
+
+  obj.AddAction("Height",
+                _("Height"),
+                _("Change the height of an object."),
+                _("the height"),
+                _("Size"),
+                "res/actions/scale24.png",
+                "res/actions/scale.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .UseStandardOperatorParameters("number")
+      .MarkAsAdvanced();
+
+  obj.AddAction("RotationAnchor",
```

Comment:
None of the property setters actions start with a "Set".
Is this a legacy thing?
Is a "Set" prefix is preferred over consistency for this extension?

Reply:
Yep that's a legacy bad decision. While we can name conditions and expressions with the name of the "property" directly, because the "get" is implied (an expression is always a getter, a condition is always a check), the actions are "verbs" (like a function changing something, which should start with a verb). 
I prefer to always use a verb, and in 90% of the cases "Set", to 1) better differentiate actions and conditions and 2) be consistent with `AddExpressionAndConditionAndAction`.

This follows the idea that conditions/actions/expressions are like this code:

```
if (something.Angle() < 30) { // A condition
  something.SetAngle( // An action
    something.Angle() + 10 // An expression
  )  
}
```

We could argue that:
- some "fluent apis", popularised by jQuery, allow to use the same function as a getter (without args) and a setter (with args), but while this is "pretty", it makes static analysis harder. Not a problem for us, but still we'll avoid this.
- if we use "Set", then we should use "Get", but we don't do that for 1) conciseness and 2) conditions are a bit more than a getter, they are a way to compare/check something.

So starting from now I'll try to enforce the "Set" for actions :)

Reply:
Ok, `SetWidth` too or this could make it hard if the action move up to Object?

Reply:
Good question.. let's keep `Width` then for consistency with the base object. And use "Set" for all "new" actions :)